packages/flutter/lib/src/widgets/async.dart

```diff
@@ -795,6 +804,9 @@ class _FutureBuilderState<T> extends State<FutureBuilder<T>> {
             _snapshot = AsyncSnapshot<T>.withError(ConnectionState.done, error, stackTrace);
           });
         }
+        if(kDebugMode && widget.debugRethrowError) {
```

Comment:
use
```dart
assert((){
}());
```
The assert will be removed in release build so it won't even cause a if check.

Reply:
Isn't this true for kDebugMode too? I'm asking this because when using kDebugMode, we can test in `flutter/test_release` [link](https://github.com/flutter/flutter/blob/42a444378f319c7ca193948eebfc81e225185c2f/packages/flutter/test_release/async_test.dart) which sets kDebugMode to false.

Reply:
kDebugMode will be false in release mode, but the code still need to do an if check.